components/sharing-dialog/src/autocomplete/sharing-autocomplete.js

```diff
@@ -0,0 +1,91 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import PropTypes from 'prop-types'
+import React, { useState, useEffect } from 'react'
+import i18n from '../locales/index.js'
+import { Autocomplete } from './autocomplete.js'
+
+const query = {
+    search: {
+        resource: 'sharing/search',
+        params: ({ search }) => ({
+            key: search,
+        }),
+    },
+}
+
+export const SharingAutocomplete = ({ selected, onSelection }) => {
```

Comment:
I think this file could be turned into a hook used by the `Autocomplete` component
* `sharing-autocomplete.js -> useAutoComplete.js`
* `autocomplete.js -> sharing-autocomplete.js`

Reply:
So judging from this line: `sharing-autocomplete.js -> useAutoComplete.js` I assume you mean that `autocomplete.js` could be a hook? Because that component is mostly markup, so to me having it be a component feels most appropriate. Or do you mean something else?

Reply:
The other way. What's currently in `sharing-autocomplete.js` is basically no markup whatsoever but using hooks.
So instead of wrapping that in a component, it could be wrapped in its own hook instead (hook logic inside hook instead of hook logic inside component).

Then the contents of what's currently in `autocomplete.js` could be moved to `sharing-autocomplete.js` and simply use the newly created hook

Reply:
Yeah I see what you mean. The reason for the current abstraction is because the `autocomplete.js` component represents the generic part. Conceptually it doesn't make as much sense to me to abstract the logic in `sharing-autocomplete.js` to a hook and then move what's in `autocomplete.js` to the `sharing-autocomplete.js` component. 

Because if you'd want another type of autocomplete you'd have to either extract the markup we moved to `sharing-autocomplete.js` to a generic component (which is what we have now), or repeat the markup.

I know it's not a situation we currently face, but that's why I tend towards the current abstraction. The other might be convenient, but to me it puts the inheritance opposite of how I conceptualize it. What do you think?

Reply:
We talked about this on slack quickly and agreed that this is mostly a matter of perspective, so I'm marking this as resolved and leave it up to @ismay what he wants to do